gravitee-apim-console-webui/src/management/api/runtime-logs-v4/runtime-logs-messages/api-runtime-logs-messages.component.spec.ts

```diff
@@ -0,0 +1,166 @@
+import { TestbedHarnessEnvironment } from '@angular/cdk/testing/testbed';
+import { UIRouterModule } from '@uirouter/angular';
+import { HttpTestingController } from '@angular/common/http/testing';
+import { MatIconTestingModule } from '@angular/material/icon/testing';
+import { NoopAnimationsModule } from '@angular/platform-browser/animations';
+
+import { ApiRuntimeLogsMessagesComponent } from './api-runtime-logs-messages.component';
+import { ApiRuntimeLogsMessgesModule } from './api-runtime-logs-messges.module';
+import { ApiRuntimeLogsMessagesHarness } from './api-runtime-logs-messages.harness';
+
+import { UIRouterState, UIRouterStateParams } from '../../../../ajs-upgraded-providers';
+import { GioUiRouterTestingModule } from '../../../../shared/testing/gio-uirouter-testing-module';
+import { fakeConnectorPlugin, fakePagedResult, MessageLog } from '../../../../entities/management-api-v2';
+import { CONSTANTS_TESTING, GioHttpTestingModule } from '../../../../shared/testing';
+import { fakeMessageLog } from '../../../../entities/management-api-v2/log/messageLog.fixture';
+import { IconService } from '../../../../services-ngx/icon.service';
+
+describe('ApiRuntimeLogsMessagesComponent', () => {
+  const API_ID = 'an-api-id';
+  const REQUEST_ID = 'a-request-id';
+  const fakeUiRouter = { go: jest.fn() };
+
+  let fixture: ComponentFixture<ApiRuntimeLogsMessagesComponent>;
+  let httpTestingController: HttpTestingController;
+  let componentHarness: ApiRuntimeLogsMessagesHarness;
+
+  const initComponent = async () => {
+    TestBed.configureTestingModule({
+      imports: [
+        ApiRuntimeLogsMessgesModule,
+        UIRouterModule.forRoot({
+          useHash: true,
+        }),
+        GioUiRouterTestingModule,
+        GioHttpTestingModule,
+        MatIconTestingModule,
+        NoopAnimationsModule,
+      ],
+      providers: [
+        { provide: UIRouterState, useValue: fakeUiRouter },
+        { provide: UIRouterStateParams, useValue: { apiId: API_ID, requestId: REQUEST_ID } },
+      ],
+    });
+
+    await TestBed.compileComponents();
+    fixture = TestBed.createComponent(ApiRuntimeLogsMessagesComponent);
+    componentHarness = await TestbedHarnessEnvironment.harnessForFixture(fixture, ApiRuntimeLogsMessagesHarness);
+    httpTestingController = TestBed.inject(HttpTestingController);
+    fixture.detectChanges();
+  };
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('GIVEN there are message logs', () => {
+    let iconServiceSpy: jest.SpyInstance;
+
+    beforeEach(async () => {
+      await initComponent();
+      iconServiceSpy = jest.spyOn(TestBed.inject(IconService), 'registerSvg').mockReturnValue('gio:mock');
+    });
+
+    it('should init the component and fetch the endpoint connector icon', async () => {
+      const messageLog = fakeMessageLog({ connectorId: 'mock', connectorType: 'ENDPOINT' });
+      expectApiWithMessageLogs([messageLog, messageLog, messageLog, messageLog, messageLog]);
```

Comment:
Maybe something like that to make explicit the length of array of the same elements ?
```suggestion
      expectApiWithMessageLogs(Array(5).fill(messageLog));
```

Reply:
this comment applies to the rest of this file

Reply:
Done